c/src/ml-api-service-extension.c

```diff
@@ -0,0 +1,1120 @@
+        break;
+      default:
+        status = ML_ERROR_INVALID_PARAMETER;
+        break;
+    }
+
+    if (status != ML_ERROR_NONE)
+      _ml_error_report_return (status,
+          "Failed to parse configuration file, cannot get the handle for pipeline node.");
+  }
+
+  return ML_ERROR_NONE;
+}
+
+/**
+ * @brief Internal function to parse pipeline info from json.
+ */
+static int
+_ml_extension_conf_parse_pipeline (ml_service_s * mls, JsonObject * pipe)
+{
+  ml_extension_s *ext = (ml_extension_s *) mls->priv;
+  int status;
+
+  if (json_object_has_member (pipe, "description")) {
+    const gchar *desc = json_object_get_string_member (pipe, "description");
+
+    status = ml_pipeline_construct (desc, NULL, NULL, &ext->pipeline);
+    if (status != ML_ERROR_NONE)
+      _ml_error_report_return (status,
+          "Failed to parse configuration file, cannot construct the pipeline.");
+  } else {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "Failed to parse configuration file, cannot get the pipeline description.");
+  }
+
+  if (json_object_has_member (pipe, "input_node")) {
+    JsonNode *node = json_object_get_member (pipe, "input_node");
+
+    status = _ml_extension_conf_parse_pipeline_node (mls, node,
+        ML_EXTENSION_NODE_TYPE_INPUT);
+    if (status != ML_ERROR_NONE)
+      _ml_error_report_return (status,
+          "Failed to parse configuration file, cannot get the input node.");
+  } else {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "Failed to parse configuration file, cannot find the input node.");
+  }
+
+  if (json_object_has_member (pipe, "output_node")) {
+    JsonNode *node = json_object_get_member (pipe, "output_node");
+
+    status = _ml_extension_conf_parse_pipeline_node (mls, node,
+        ML_EXTENSION_NODE_TYPE_OUTPUT);
+    if (status != ML_ERROR_NONE)
+      _ml_error_report_return (status,
+          "Failed to parse configuration file, cannot get the output node.");
+  } else {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "Failed to parse configuration file, cannot find the output node.");
+  }
+
+  status = ml_pipeline_start (ext->pipeline);
+  if (status != ML_ERROR_NONE)
+    _ml_error_report_return (status,
+        "Failed to parse configuration file, cannot start the pipeline.");
+
+  return ML_ERROR_NONE;
+}
+
+/**
+ * @brief Internal function to parse app information from json.
+ */
+static int
+_ml_extension_conf_parse_information (ml_service_s * mls, JsonObject * info)
+{
+  ml_extension_s *ext = (ml_extension_s *) mls->priv;
+  JsonObjectIter iter;
+  JsonNode *node;
+  const gchar *name, *value;
+  int status;
+
+  status = ml_option_create (&ext->conf_info);
+  if (status != ML_ERROR_NONE)
+    _ml_error_report_return (status,
+        "Failed to parse configuration file, cannot create ml-option handle.");
+
+  json_object_iter_init (&iter, info);
+  while (json_object_iter_next (&iter, &name, &node)) {
```

Comment:
Still don't know why, however, Using json_glib apis of `..._ordered` will fix the tizen unittest fails.
```suggestion
  json_object_iter_init_ordered (&iter, info);
  while (json_object_iter_next_ordered (&iter, &name, &node)) {
```

Or we could use `json_object_get_object ...` rather than `_iter_` APIs

Reply:
updated, thanks.